### PR TITLE
Image Pipeline: read gz files, handle 1M cells, add CLI arguments (SCP-4641)

### PR DESCRIPTION
**Pull request by eweitz**
This enables Image Pipeline to read gzipped files produced by the draft expression scatter data pipeline.  

Files will ultimately be decompressed in transit, we presume.  Decompressing them on the client is mostly a way to pick up draft scatter data pipeline output, until that other pipeline writes out individual gzipped files (without tarballing) and sets appropriate GCS metadata on them to enable [decompressive transcoding](https://cloud.google.com/storage/docs/transcoding#decompressive_transcoding).

This also works around an issue in #1605 with using scatter files for the 1.3M cell study.  Using scatter files failed for that study, but not the 49k DE pilot study.  This seems caused not by the scatter data pipeline, but rather with using trimmed URLs to conditionally fetch only expression arrays (and not also annotation, cell, and coordinate) data in Image Pipeline.  The workaround still uses local data, but more than necessary, and can be optimized as a later, separate task.

To test:
* Install dependencies, if you hadn't in previous PR: `cd image-pipeline; yarn install`
* Find the accession for your local "Human milk -- differential expression" study
  * If you don't have that DE pilot study, you can add it via Terminal like so: `echo "SyntheticStudyPopulator.populate('human_milk_de_49k_cell')" | bundle exec rails console -e development`
* Confirm "progressive_loading" feature flag is _disabled_ for the study
* Download [All_Cells_UMAP_data.tar.gz](https://console.cloud.google.com/storage/browser/_details/broad-singlecellportal-public/test/All_Cells_UMAP_data.tar.gz), untar it, change untarred directory name from "All_Cells_UMAP-3d16b946-86a7-4d50-bce8-b97e2ab02734" to "All_Cells_UMAP"
* Ensure Rails is running (`rails s`)
* Changing "SCP138" to your DE pilot study accession and the `--json-dir` value to the path of your "All_Cells_UMAP" directory above, run:
`export NODE_TLS_REJECT_UNAUTHORIZED=0; time node expression-scatter-plots.js --accession="SCP138" --environment="development" --cores="1" --cluster="All_Cells_UMAP" --json-dir="/Users/eweitz/Downloads/All_Cells_UMAP/" --debug; tput bel # localhost, human milk DE study`
  * If you immediately see `TypeError: fetch failed`, edit `expression-scatter-plots.js` to use `https://127.0.0.1:3000` instead of `https://localhost:3000`, then re-run above
* Confirm `ls -latr output/<your DE pilot study accession>/images/`
  * If above times out, then simply try running above `expression-scatter-plots.js` command again

This satisfies SCP-4641.

### CODE REVIEW

File: image-pipeline/expression-scatter-plots.js [hunk 1]
```diff
@@ -7,30 +7,42 @@
...
 
 // TODO (SCP-4564): Document how to adjust network rules to use staging
-// const origin = 'https://singlecell-staging.broadinstitute.org'
-const origin = 'https://localhost:3000'
+const originsByEnvironment = {
+  'development': 'https://localhost:3000',
+  'staging': 'https://singlecell-staging.broadinstitute.org',
+  'production': 'https://singlecell.broadinstitute.org'
+}
+const environment = values.environment ?? 'development'
```
Comment (ehanna4) [inline image-pipeline/expression-scatter-plots.js:44]: `??` only looks at explicitly `null` or `undefined` so an empty string would get through in this case. Is that a reasonable value for `environment` to be set to here, or would it better to check for any falsy and do `const environment = values.environment || 'development'` ?

File: image-pipeline/expression-scatter-plots.js [hunk 2]
```diff
@@ -7,30 +7,42 @@
...
+import { gunzipSync, strFromU8 } from 'fflate'
 import puppeteer from 'puppeteer'
-import { exit } from 'node:process'
+import sharp from 'sharp'
 
 const args = process.argv.slice(2)
 
 const options = {
-  accession: { type: 'string' }
+  'accession': { type: 'string' }, // SCP accession
```
Comment (ehanna4) [inline image-pipeline/expression-scatter-plots.js:22]: Appreciate the comments here!

**Review by ehanna4 — APPROVED** [review]
Functional review: works as advertised
code review: left one non-blocking comment/suggestion

**Review by jlchang — APPROVED** [review]
Functional review - watched expression-scatter-plots.js render gene expression plots for several genes in-browser but only against staging. Something about using localhost is blocked when I try in my development environment (Note that I can't get vite-dev to communicate with my local instance so that may be part of the problem)

File: image-pipeline/expression-scatter-plots.js (hunk 1 above)
Comment (eweitz) [reply image-pipeline/expression-scatter-plots.js]: Nice catch!  Done in 3472483.

**Pull request merged**